src/web/app/components/course-edit-form/course-edit-form.component.html

```diff
@@ -0,0 +1,122 @@
+    <strong *ngIf="isInEditMode">Course</strong>
+    <strong *ngIf="isInAddMode">Create a new Course</strong>
+    <div class="card-header-btn-toolbar">
+        <ng-container *ngIf="isInEditMode">
+            <button id="btn-edit-course" class="btn btn-primary btn-sm" [disabled]="!canModifyCourse" *ngIf="!isEditing"
+                (click)="setIsEditing($event)">
+                <i class="fa fa-pen" aria-hidden="true"></i>
+                Edit
+            </button>
+            <button id="btn-cancel-course" *ngIf="isEditing" class="btn btn-primary btn-sm" [disabled]="isSaving"
+                (click)="discardChangesHandler()">
+                <i class="fa fa-times" aria-hidden="true"></i>
+                Cancel
+            </button>
+            <button id="btn-delete-course" class="btn btn-primary btn-sm"
+                ngbTooltip="Delete the course and its corresponding students and sessions"
+                [disabled]="!canModifyCourse || isSaving" (click)="deleteCourseHandler()">
+                <i class="fa fa-trash" aria-hidden="true"></i>
+                Delete
+            </button>
+        </ng-container>
+        <ng-container *ngIf="isInAddMode">
+            <button id="btn-edit-course" class="btn btn-primary btn-sm" [disabled]="!canModifyCourse"
+                (click)="copyCourseHandler()" [disabled]="isCopying || isInstitutesArrayEmpty">
+                <i class="fa fa-copy" aria-hidden="true" *ngIf="!isCopying"></i>
+                <tm-ajax-loading *ngIf="isCopying"></tm-ajax-loading>
+                Copy from previous courses
+            </button>
+            <button id="btn-close-form" class="btn btn-primary btn-sm" (click)="closeFormHandler()">
+                <i class="fa fa-times" aria-hidden="true"></i>
+            </button>
+        </ng-container>
+    </div>
+</div>
+<div class="card-body fill-plain">
+    <form #courseForm="ngForm" (ngSubmit)="submitHandler()">
+        <div class="form-row form-group">
+            <label class="col-sm-2 font-weight-bold text-md-right">Course ID:</label>
+            <div class="col-sm-9">
+                <input id="course-id" name="courseId" class="col-xs-12 form-control" type="text" tabindex="1"
+                    placeholder="e.g. CS3215-2013Semester1" #courseId="ngModel" [(ngModel)]="model.course.courseId"
+                    [disabled]="isInEditMode || isInputDisabled" [maxlength]="FormValidator.COURSE_ID_MAX_LENGTH"
+                    required>
+                <div [hidden]="courseId.valid || (courseId.pristine && courseId.untouched) || isSaving || isDisplayOnly"
+                    class="invalid-field">
+                    <i class="fa fa-exclamation-circle" aria-hidden="true"></i>
+                    The field Course ID should not be empty.
+                </div>
+            </div>
+        </div>
+        <div class="form-row form-group">
+            <label class="col-sm-2 font-weight-bold text-md-right">Course Name:</label>
```

Comment:
The style `font-weight-bold text-md-right` looks used, can consider making it a class

Reply:
@FergusMok May I clarify whether you mean make `font-weight-bold text-md-right` into a single custom class (since they are already bootstrap classes)?

Reply:
@thikhinab With SCSS you're able to use @extend for multiple bootstrap classes to be put under one class. However, in hindsight I realize this might hamper readability and might bring much benefit. Sorry, you can choose to disregard this message 🙏

Reply:
@FergusMok Understood, thanks for suggesting this. Will look if there are other places I can `@extend` classes.